src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryCustomImpl.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.justice.digital.hmpps.digitalprisonreportingmi.data
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.jdbc.core.JdbcTemplate
+import uk.gov.justice.digital.hmpps.digitalprisonreportingmi.model.ExternalMovement
+import uk.gov.justice.digital.hmpps.digitalprisonreportingmi.model.ExternalMovementFilter
+import java.time.LocalDate
+import java.time.LocalTime
+
+class ExternalMovementRepositoryCustomImpl : ExternalMovementRepositoryCustom {
+  @Autowired
+  lateinit var jdbcTemplate: JdbcTemplate
+  override fun list(selectedPage: Long, pageSize: Long, sortColumn: String, sortedAsc: Boolean, filters: Map<ExternalMovementFilter, Any>): List<ExternalMovement> {
```

Comment:
I'm uneasy about putting strings from the REST endpoint into the SQL query. Have all of these been validated and/or checked that they fall within an enumeration of expected values?

Reply:
I am planning to change these to use something like a prepared statement to avoid this issue or validate them in a different way. I just wanted to construct the query first to be able to test it.